File: src/main/java/top/zbeboy/isy/service/data/CollegeApplicationServiceImpl.java
package top.zbeboy.isy.service.data;

import org.jooq.DSLContext;
import org.jooq.Result;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Propagation;
import org.springframework.transaction.annotation.Transactional;
import top.zbeboy.isy.domain.tables.pojos.CollegeApplication;
import top.zbeboy.isy.domain.tables.records.CollegeApplicationRecord;

import static top.zbeboy.isy.domain.Tables.COLLEGE_APPLICATION;

/**
 * Created by lenovo on 2016-10-05.
 */
@Service("collegeApplicationService")
@Transactional(propagation = Propagation.SUPPORTS, readOnly = true)
public class CollegeApplicationServiceImpl implements CollegeApplicationService {

    private final Logger log = LoggerFactory.getLogger(CollegeApplicationServiceImpl.class);

    private final DSLContext create;

    @Autowired
    public CollegeApplicationServiceImpl(DSLContext dslContext) {
        this.create = dslContext;
    }

    @Override
    public void deleteByApplicationId(int applicationId) {
        create.deleteFrom(COLLEGE_APPLICATION)
                .where(COLLEGE_APPLICATION.APPLICATION_ID.eq(applicationId))
                .execute();
    }

    @Override
    public Result<CollegeApplicationRecord> findByCollegeId(int collegeId) {
        return create.selectFrom(COLLEGE_APPLICATION)
                .where(COLLEGE_APPLICATION.COLLEGE_ID.eq(collegeId))
                .fetch();
    }

    @Transactional(propagation = Propagation.REQUIRED, readOnly = false)
    @Override
    public void save(CollegeApplication collegeApplication) {
        create.insertInto(COLLEGE_APPLICATION)
                .set(COLLEGE_APPLICATION.COLLEGE_ID, collegeApplication.getCollegeId())
                .set(COLLEGE_APPLICATION.APPLICATION_ID, collegeApplication.getApplicationId())
                .execute();
    }

    @Override
    public void deleteByCollegeId(int collegeId) {
        create.deleteFrom(COLLEGE_APPLICATION)
                .where(COLLEGE_APPLICATION.COLLEGE_ID.eq(collegeId))
                .execute();
    }
}

File: src/main/java/top/zbeboy/isy/service/data/CollegeApplicationServiceImpl.kt
package top.zbeboy.isy.service.data

import org.jooq.DSLContext
import org.jooq.Result
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Propagation
import org.springframework.transaction.annotation.Transactional
import top.zbeboy.isy.domain.tables.pojos.CollegeApplication
import top.zbeboy.isy.domain.tables.records.CollegeApplicationRecord

import top.zbeboy.isy.domain.Tables.COLLEGE_APPLICATION
/**
 * Created by zbeboy 2017-12-02 .
 **/
@Service("collegeApplicationService")
@Transactional(propagation = Propagation.SUPPORTS, readOnly = true)
open class CollegeApplicationServiceImpl @Autowired constructor(dslContext: DSLContext) :CollegeApplicationService {

    private val create: DSLContext = dslContext

    override fun deleteByApplicationId(applicationId: String) {
        create.deleteFrom<CollegeApplicationRecord>(COLLEGE_APPLICATION)
                .where(COLLEGE_APPLICATION.APPLICATION_ID.eq(applicationId))
                .execute()
    }

    override fun findByCollegeId(collegeId: Int): Result<CollegeApplicationRecord> {
        return create.selectFrom<CollegeApplicationRecord>(COLLEGE_APPLICATION)
                .where(COLLEGE_APPLICATION.COLLEGE_ID.eq(collegeId))
                .fetch()
    }

    @Transactional(propagation = Propagation.REQUIRED, readOnly = false)
    override fun save(collegeApplication: CollegeApplication) {
        create.insertInto<CollegeApplicationRecord>(COLLEGE_APPLICATION)
                .set(COLLEGE_APPLICATION.COLLEGE_ID, collegeApplication.collegeId)
                .set(COLLEGE_APPLICATION.APPLICATION_ID, collegeApplication.applicationId)
                .execute()
    }

    override fun deleteByCollegeId(collegeId: Int) {
        create.deleteFrom<CollegeApplicationRecord>(COLLEGE_APPLICATION)
                .where(COLLEGE_APPLICATION.COLLEGE_ID.eq(collegeId))
                .execute()
    }
}